frontends/main/src/page-components/ResourceCard/ResourceCard.tsx

```diff
@@ -54,7 +55,10 @@ const useResourceCard = (resource?: LearningResource | null) => {
       }
     }, [user])
 
+  const onClick = useLearningResourceDetailSetCache(resource)
```

Comment:
I considered various possible ways to pre-populate the `useLearningResourceDetail` data, e.g., 
- inside `useLearningResourceDetail` hook, search the query cache for the resource
    - hard because the query cache stores lots of different shapes of data, and we generally don't know during search whether something is a resource
- prepopulate the details cache whenever we make a listing call
    - hard because there are so many different listing calls

In the end, setting it here seemed easiest. We have many different versions of "listings" (featured courses, various other carousels, search results, userlist resources...) but generally it's pretty rare that we use a details. And currently the only time we do, it is triggered via the resource cards.

Reply:
bonus side effect... drawer loads way faster for everyone!